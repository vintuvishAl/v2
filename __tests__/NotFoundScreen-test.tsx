import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';

import NotFoundScreen from '@/app/+not-found';

// Mock expo-router
jest.mock('expo-router', () => ({
  Link: ({ children, href, ...props }: any) => <div {...props}>{children}</div>,
  Stack: { Screen: ({ options }: any) => <div>{options?.title}</div> },
}));

describe('<NotFoundScreen />', () => {
  test('renders error message', () => {
    const { getByText } = render(<NotFoundScreen />);
    expect(getByText('This screen does not exist.')).toBeTruthy();
  });

  test('renders link to home screen', () => {
    const { getByText } = render(<NotFoundScreen />);
    expect(getByText('Go to home screen!')).toBeTruthy();
  });

  test('link is pressable', () => {
    const { getByText } = render(<NotFoundScreen />);
    const link = getByText('Go to home screen!');
    
    fireEvent.press(link);
    expect(link).toBeTruthy();
  });

  test('has proper styling classes', () => {
    const { getByText } = render(<NotFoundScreen />);
    const title = getByText('This screen does not exist.');
    const link = getByText('Go to home screen!');
    
    expect(title).toBeTruthy();
    expect(link).toBeTruthy();
  });

  test('matches snapshot', () => {
    const tree = render(<NotFoundScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
