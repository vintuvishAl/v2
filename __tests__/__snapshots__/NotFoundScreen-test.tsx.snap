// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundScreen /> matches snapshot 1`] = `
[
  <div>
    Oops!
  </div>,
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#111827",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#f3f4f6",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 20,
        }
      }
    >
      This screen does not exist.
    </Text>
    <div
      style={
        {
          "marginTop": 15,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "color": "#3b82f6",
            "fontSize": 16,
            "textDecorationLine": "underline",
          }
        }
      >
        Go to home screen!
      </Text>
    </div>
  </View>,
]
`;
